### PR TITLE
Dual Specular for Skin

### PR DESCRIPTION
**Pull request by antonmic**
1. Adds dual specular for skin and eye materials
2. Cleaned up skin.materialtype file to use PropertyGroup.json files instead of duplicating settings
3. Fixed custom debug floats not being passed from the RenderDebugComponent down to the SRG
4. Added more custom debug floats for debugging and changed hard limits to soft limits to allow greater flexibility
5. Add #define MATERIAL_HAS_UVS 0 to material .azsli files that don't have UVs. This allows custom material pipelines that assume the existence of UVs in vertex and pixel geometry structs to handle this case instead of causing shader compilation failures.

### CODE REVIEW

File: Gems/Atom/Feature/Common/Assets/Materials/ReflectionProbe/ReflectionProbeVisualization.azsli [hunk 1]
```diff
@@ -15,6 +15,8 @@
 
 #if MATERIALPIPELINE_SHADER_HAS_PIXEL_STAGE
     
+#define MATERIAL_HAS_UVS 0
```
Comment (AMZN-alexpete) [inline Gems/Atom/Feature/Common/Assets/Materials/ReflectionProbe/ReflectionProbeVisualization.azsli:18]: Where is the logic that uses `MATERIAL_HAS_UVS`?

File: Gems/Atom/Feature/Common/Assets/Shaders/Materials/MaterialInputs/DualSpecular.azsli [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                          out float outDualSpecRoughness, out float outDualSpecRoughnessA2, out float outDualSpecF0, out float outDualSpecFactor)
+{
+    float dummyRoughnessA = 0;
+    outDualSpecFactor = inDualSpecFactor;
+    outDualSpecRoughness = pow(roughnessLinear, dualSpecRoughnessExponent) * dualSpecRoughnessMultiplier;
+    CalculateRoughnessValues(normal, outDualSpecRoughness, dummyRoughnessA, outDualSpecRoughnessA2);
+    outDualSpecF0 = specularF0 * dualSpecF0Multiplier;
+
+    if(!o_enableDualSpecular)
+        outDualSpecFactor = 0;
```
Comment (AMZN-alexpete) [inline Gems/Atom/Feature/Common/Assets/Shaders/Materials/MaterialInputs/DualSpecular.azsli:32]:
```suggestion
    if(!o_enableDualSpecular)
    {
        outDualSpecFactor = 0;
    }
```

File: Gems/Atom/Feature/Common/Assets/Shaders/Materials/Skin/Skin_LightingBrdf.azsli [hunk 3]
```diff
@@ -28,17 +32,20 @@
...
     return diffuse;
 }
+
+
+real3 GetSpecularLighting_Skin(Surface surface, LightingData lightingData, const real3 lightIntensity, const real3 dirToLight)
+{
+    real3 specular = SpecularGGX(lightingData.dirToCamera, dirToLight, surface.normal, surface.specularF0, lightingData.NdotV, surface.roughnessA2, lightingData.multiScatterCompensation);
+
+    real3 dualSpecular = SpecularGGX(lightingData.dirToCamera, dirToLight, surface.normal, surface.dualSpecF0.xxx, lightingData.NdotV, surface.dualSpecRoughnessA2, lightingData.multiScatterCompensation);
+    specular = lerp(specular, dualSpecular, surface.dualSpecFactor);
```
Comment (AMZN-alexpete) [inline Gems/Atom/Feature/Common/Assets/Shaders/Materials/Skin/Skin_LightingBrdf.azsli:46]: should we wrap this in `if(o_enableDualSpecular)`?

**Review by AMZN-alexpete — COMMENTED** [review]
Just a couple questions, may also want to mention in the PR that this adds Dual specular for the Eye material as well.

File: Gems/Atom/Feature/Common/Assets/Materials/Pipelines/MainPipeline/ForwardPass_SkinLighting.azsli [hunk 4]
```diff
@@ -16,6 +16,8 @@
 #define ENABLE_ANISOTROPY 0
 #define ENABLE_SHADER_DEBUGGING 1
 #define ENABLE_CLEAR_COAT 0
+#define ENABLE_DUAL_SPECULAR 1
+#define FORCE_IBL_IN_FORWARD_PASS 1
```
Comment (moudgils) [inline Gems/Atom/Feature/Common/Assets/Materials/Pipelines/MainPipeline/ForwardPass_SkinLighting.azsli:20]: Do we want this for main pipeline? I thought this was enabled for LowEnd pipeline

File: Gems/Atom/Feature/Common/Assets/Materials/ReflectionProbe/ReflectionProbeVisualization.azsli (hunk 1 above)
Comment (antonmic) [reply Gems/Atom/Feature/Common/Assets/Materials/ReflectionProbe/ReflectionProbeVisualization.azsli:18]: In non-public code 
The usage is essentially
MyCustomShader.azsli:
#if MATERIAL_HAS_UVS
    float2 uv = vertexData.uv;
#else
    float2 uv = float2(0, 0);
#endif

File: Gems/Atom/Feature/Common/Assets/Shaders/Materials/MaterialInputs/DualSpecular.azsli (hunk 2 above)
Comment (antonmic) [reply Gems/Atom/Feature/Common/Assets/Shaders/Materials/MaterialInputs/DualSpecular.azsli:32]: Good point, will fix

Comment (moudgils) [reply Gems/Atom/Feature/Common/Assets/Shaders/Materials/MaterialInputs/DualSpecular.azsli:32]: Yeah. Let hide this code behind the shader option so it gets baked out.

File: Gems/Atom/Feature/Common/Assets/Shaders/Materials/Skin/Skin_LightingBrdf.azsli [hunk 5]
```diff
@@ -28,17 +32,20 @@
...
-    {
-        // Use diffuse brdf containing double Fresnel (enter/exit surface) terms if subsurface scattering is enabled
-        diffuse = NormalizedDisneyDiffuse(surface.albedo, surface.normal, lightingData.dirToCamera, dirToLight, surface.roughnessLinear);
-    }
-    else
-    {
-        diffuse = DiffuseLambertian(surface.albedo, surface.normal, dirToLight, lightingData.diffuseResponse);
-    }
-    
+    real3 diffuse = NormalizedDisneyDiffuse(surface.albedo, surface.normal, lightingData.dirToCamera, dirToLight, surface.roughnessLinear);
```
Comment (moudgils) [inline Gems/Atom/Feature/Common/Assets/Shaders/Materials/Skin/Skin_LightingBrdf.azsli:35]: Any reason to use Disney all the time? Just better visual quality?

Comment (AMZN-alexpete) [reply Gems/Atom/Feature/Common/Assets/Shaders/Materials/Skin/Skin_LightingBrdf.azsli:35]: when in doubt, choose the mouse🥁

**Review by AMZN-alexpete — APPROVED** [review]

**Review by moudgils — APPROVED** [review]

**Pull request merged**